### PR TITLE
all changes to make tcp-router bpm

### PR DESCRIPTION
**Pull request by zachgersh**
[#158365507]

### CODE REVIEW

File: jobs/tcp_router/templates/bpm.yml.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  - -tcpLoadBalancerBaseConfig
+  - "/var/vcap/jobs/tcp_router/config/haproxy.conf.template"
+  - -config
+  - "/var/vcap/jobs/tcp_router/config/tcp_router.yml"
+  - -tcpLoadBalancerStatsUnixSocket
+  - "/var/vcap/data/tcp_router/config/haproxy.sock"
+  - -statsCollectionInterval
+  - <%= p("tcp_router.tcp_stats_collection_interval") %>
+  - -haproxyReloader
+  - "/var/vcap/jobs/tcp_router/bin/noop"
```
Comment (rosenhouse) [inline jobs/tcp_router/templates/bpm.yml.erb]: why not [`true`](http://man7.org/linux/man-pages/man1/true.1.html) ?

Comment (zachgersh) [reply jobs/tcp_router/templates/bpm.yml.erb]: could do that as well - great suggestion!

File: jobs/tcp_router/templates/killer [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -eo pipefail
+
+PID=$(/var/vcap/jobs/bpm/bin/bpm pid tcp_router -p haproxy)
+
+echo "going to kill ${PID}"
+kill -USR1 "${PID}"
+echo "killed"
```
Comment (rosenhouse) [inline jobs/tcp_router/templates/killer]: is this script really a `killer` or just a `config_reloader`?  as someone skimming things from a high-level, I might get the mistaken impression that this script terminates the process.

Comment (zachgersh) [reply jobs/tcp_router/templates/killer]: agreed - name change would clarify its intent.

File: jobs/tcp_router/templates/bpm-pre-start.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
```
Comment (aaronshurley) [inline jobs/tcp_router/templates/bpm-pre-start.erb]: Is there a reason for a bpm pre-start vs a bosh pre-start?

File: jobs/tcp_router/monit [hunk 4]
```diff
@@ -1,7 +1,17 @@
...
-    as uid vcap and gid vcap
+  with pidfile /var/vcap/sys/run/bpm/tcp_router/tcp_router.pid
+  start program "/var/vcap/jobs/bpm/bin/bpm start tcp_router"
+  stop program "/var/vcap/jobs/bpm/bin/bpm stop tcp_router"
   group vcap
+  depends on haproxy
+
+check file haproxy_config
+  with path /var/vcap/data/tcp_router/config/haproxy.conf
+  if changed timestamp then exec "/bin/bash -c '/var/vcap/jobs/tcp_router/bin/killer &>/var/vcap/data/tcp_router/config/killer.out'"
```
Comment (aaronshurley) [inline jobs/tcp_router/monit]: why not log to `/var/vcap/sys/log/tcp_router/config_reloader.log` to be consistent with log location?

**Comment by zachgersh** [comment]
closing but not deleting the branch as of yet.

**Pull request closed without merging**